feature-libs/asm/customer-360/components/sections/asm-customer-coupon/asm-customer-coupon.component.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  Customer360Coupon,
+  Customer360CouponList,
+  Customer360Facade,
+  Customer360Type,
+} from '@spartacus/asm/customer-360/root';
+import { UserIdService } from '@spartacus/core';
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  combineLatest,
+  of,
+} from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { ActiveCartFacade, CartVoucherFacade } from '@spartacus/cart/base/root';
+import { Customer360SectionContext } from '../customer-360-section-context.model';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'cx-asm-customer-coupon',
+  templateUrl: './asm-customer-coupon.component.html',
+})
+export class AsmCustomerCouponComponent implements OnInit, OnDestroy {
+  showErrorAlert$ = new BehaviorSubject<boolean>(false);
+  showErrorAlertForApplyAction$ = new BehaviorSubject<boolean>(false);
+  currentCartId: string | undefined;
+  userId: string;
+  entries$: Observable<Array<Customer360Coupon>>;
+  subscription = new Subscription();
+
+  constructor(
+    protected context: Customer360SectionContext<Customer360CouponList>,
+    protected cartVoucherService: CartVoucherFacade,
+    protected userIdService: UserIdService,
+    protected activeCartFacade: ActiveCartFacade,
+    protected customer360Facade: Customer360Facade
+  ) {}
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.userIdService.getUserId().subscribe((user) => {
+        this.userId = user ?? '';
+      })
+    );
+    this.subscription.add(
+      this.activeCartFacade.requireLoadedCart().subscribe((cart) => {
+        this.currentCartId = cart?.code;
+      })
+    );
+    this.subscription.add(
+      this.cartVoucherService.getAddVoucherResultError().subscribe((error) => {
+        if (error) {
+          this.refreshComponent();
+          this.showErrorAlertForApplyAction$.next(true);
+        }
+      })
+    );
+    this.showErrorAlert$.next(false);
+    this.showErrorAlertForApplyAction$.next(false);
+    this.fetchCoupons();
+  }
+
+  fetchCoupons() {
```

Comment:
Please add public/protected before routines. Also returns of them.
```suggestion
  public fetchCoupons(): Array<Customer360Coupon>  {
```

Reply:
this function returns void. `public` will be added.